### PR TITLE
[IOTDB-4206] Record DataNode's read-only status in ConfigNode

### PR DESCRIPTION
**Pull request by CRZbulabula**
The DataNode has status ReadOnly which indicates itself has persistence problem. It's the duty of ConfigNode-leader to monitor DataNodes' running status and take corrective measures when some DataNodes run abnormally. The ConfigNode-leader will have the ability to monitor DataNode's ReadOnly and Error status after the following updates:

1. Merge SystemStatus into NodeStatus
2. DataNodes will report their status during each heartbeat
3. ConfigNode-leader will broadcast the latest RegionRouteMap as soon as some DataNodes change their status.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java [hunk 1]
```diff
@@ -384,6 +384,7 @@ public TSStatus invalidateSchemaCache(TInvalidateCacheReq req) {
   public THeartbeatResp getDataNodeHeartBeat(THeartbeatReq req) throws TException {
     THeartbeatResp resp = new THeartbeatResp();
     resp.setHeartbeatTimestamp(req.getHeartbeatTimestamp());
+    resp.setStatus(IoTDBDescriptor.getInstance().getConfig().getSystemStatus().getStatus());
```
Comment (Beyyes) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:387]: `getNodeStatus` is better?

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/ConfigNodeHeartbeatCache.java [hunk 2]
```diff
@@ -60,23 +60,15 @@ public boolean updateLoadStatistic() {
...
-    NodeStatus originStatus;
-    switch (status) {
-      case Running:
-        originStatus = NodeStatus.Running;
-        break;
-      case Unknown:
-      default:
-        originStatus = NodeStatus.Unknown;
-    }
+    String originStatus = status.getStatus();
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/ConfigNodeHeartbeatCache.java:63]: what's the difference between `Unknown` and `Error`?
`Unknown` -> `Error` also need to update?

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java [hunk 3]
```diff
@@ -94,15 +84,8 @@ public long getLoadScore() {
...
-    switch (status) {
-      case Running:
-        return loadScore;
-      case Unknown:
-      default:
-        // The Unknown Node will get the highest loadScore
-        return Long.MAX_VALUE;
-    }
+    // The DataNode whose status isn't Running will get the highest loadScore
+    return status == NodeStatus.Running ? loadScore : Long.MAX_VALUE;
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java:88]: same impl to the `getLoadScore` of `ConfigNodeHeartbeatCache`.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java [hunk 4]
```diff
@@ -112,12 +95,6 @@ public NodeStatus getNodeStatus() {
...
 
     // Return a copy of status
-    switch (status) {
-      case Running:
-        return NodeStatus.Running;
-      case Unknown:
-      default:
-        return NodeStatus.Unknown;
-    }
+    return NodeStatus.valueOf(status.getStatus());
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java:98]: same impl to the `getNodeStatus` of `ConfigNodeHeartbeatCache`.

image

**Review by Beyyes — COMMENTED** [review]
Change the variable naming of `NodeStatus` to CAPITAL and UNDERLINE when next pr.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java (hunk 4 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java:98]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java (hunk 3 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/DataNodeHeartbeatCache.java:90]: Fixed~

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java (hunk 1 above)
Comment (CRZbulabula) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java]: Fixed~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/ConfigNodeHeartbeatCache.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/load/heartbeat/ConfigNodeHeartbeatCache.java:63]: Fixed. We only need to broadcast the latest RegionRouteMap when Running changes to another state or when other states change to Running.

**Review by Beyyes — APPROVED** [review]

**Review by HeimingZ — APPROVED** [review]
LGTM :)

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**